nodebuilder/state/core.go

```diff
@@ -20,6 +20,11 @@ func coreAccessor(
 	sync *sync.Syncer[*header.ExtendedHeader],
 	fraudServ libfraud.Service,
 ) (*state.CoreAccessor, *modfraud.ServiceBreaker[*state.CoreAccessor]) {
+	if corecfg == core.DefaultConfig() {
```

Comment:
Could you check for the exact fields that are missing? This feels wrong and fragile.